### DISCUSSION TITLE
Start a V3 branch, merge all V3 PRs there (and start cutting alpha/pre-release gems?)

### DISCUSSION THREAD

**Discussion by stevegeek** [rswag/rswag#678] [General]
I would love to get onto the latest of RSwag changes that are lined up for V3, (ie https://github.com/rswag/rswag/milestone/2) , could one start a branch into which all these are merged, and then from there some pre-release V3 gems can be cut? 

This way it should be easier to start testing the latest changes for V3, and new contributions will be easier as changes could be make directly against this future version (and PRs based off that branch)

Happy to help if needed.

**Comment by mattpolito** [comment]
At this point, that is potentially a good idea. The cleanup PR #574 is ready but I feel a tad aggressive of a change. Haven't heard from @romanblanco since I commented so it's probably time to just start addressing it myself. 

Are there specific changes you're looking for regarding v3?

**Reply by stevegeek** [reply]
Thanks @mattpolito :) I am looking primarily to have my changes from https://github.com/romanblanco/rswag/pull/1 incorporated. This PR is a rebase of my work on #574 (original PR https://github.com/rswag/rswag/pull/656). 

However any other fixes earmarked for v3 will also be appreciated.

**Reply by romanblanco** [reply]
Hey @stevegeek, @mattpolito, sorry for the silence. I have been busy lately and couldn't find time to participate here.

I like the idea of starting a branch for V3 and merging [relevant changes](https://github.com/rswag/rswag/milestone/2) in it. Will do my best to do it this week.

**Reply by romanblanco** [reply]
@stevegeek The `v3` branch is now in the repository: https://github.com/rswag/rswag/tree/v3.

I'll soon start going through PRs that are ready and merge them into the branch.

**Reply by stevegeek** [reply]
Great thanks !

**Reply by romanblanco** [reply]
@stevegeek I've merged https://github.com/rswag/rswag/pull/574 into the `v3` branch now, should unblock you from your work :wink:

**Reply by romanblanco** [reply]
@BookOfGreg @jtannas, could you help set branch https://github.com/rswag/rswag/tree/v3 as protected?

**Reply by jtannas** [reply]
@BookOfGreg I don't have access to the branch protection settings so going to need you for this one

**Reply by romanblanco** [reply]
https://github.com/rswag/rswag/pull/714

**Reply by stevegeek** [reply]
@romanblanco sorry I didn't get back on the PRs , I have been busy. Will try get the review tweaks done asap

**Reply by BookOfGreg** [reply]
Added protection and requirement to PR to `v3` branch. Is that what you wanted?

**Reply by romanblanco** [reply]
Great! Thank you, @BookOfGreg

**Comment by pkondzior** [comment]
Hello, what are the plans for 3.0 release ? From the milestone perspective we are 60% away but it seems that important pull requests are available and close to being merged. Is there any contributor help required to push this forward ?

**Reply by romanblanco** [reply]
@pkondzior I'm currently busy, but working on getting back to finalize the 3.0 release.

**Comment by valdis** [comment]
@stevegeek @romanblanco Hey guys i was looking into adding support for webhooks to rswag, but there are v3, v3_updata and other branches floating in the air. How about we merge them and release as a relese candidate version? 

What else needs to be done to finalize the 3.0 release?

**Reply by stevegeek** [reply]
Agreed, I would love it if we could do this

**Reply by romanblanco** [reply]
@valdis, I'd love to include @stevegeek's changes in https://github.com/rswag/rswag/pull/723 into the v3 release.

Once that is finished, I'll start wrapping up https://github.com/rswag/rswag/pull/714 and release the rswag release candidate version.

**Reply by stevegeek** [reply]
Ok ill hopefully get to it later today!

**Comment by totrash** [comment]
Hey @romanblanco
Is there any chance for a v3 release or at least 2.15 release from master
I would love to have that new option "openapi_all_properties_required" available which is already added to the master

**Reply by romanblanco** [reply]
@totrash, I plan to do 2.15 release before v3. Expecting to do it this week.

Wanted to get https://github.com/rswag/rswag/pull/785 into 2.15 as well.

**Comment by romanblanco** [comment]
I feel like what should've land in v3 is now in and we could release a pre-release version.

**Reply by stevegeek** [reply]
Great!
